### PR TITLE
feat: ban-user command

### PR DESCRIPTION
**Pull request by berry-13**
### created a command `npm run ban-user` to ban a specific user "theoretically" permanently

image

## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Testing

1. controllo da diverse browser con diversi IP se l'utente è stato bannato e che non può inviare piu messaggi

NOTE: When the account is banned, if I restart the server I can log back in but the moment I write a message and send it, it kicks me out and doesn't send the message

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

**Comment by danny-avila** [comment]
why not also allow a user input for the duration?

**Comment by Berry-13** [comment]
> Why not also allow a user input for the duration?

Because I can't find a way to specify the time

**Comment by techwithanirudh** [comment]
> > Why not also allow a user input for the duration?
> 
> Because I can't find a way to specify the time

```
 const errorMessage = {
    type: 'concurrent',
    violation_count: 20,
    user_id: user._id,
    prev_count: 0,
    duration: 1000 * 60 * 60 * 24 * 7,
  };

  await banViolation(req, res, errorMessage);
```

You're specifying the duration here, I believe.

**Comment by Berry-13** [comment]
> > > Why not also allow a user input for the duration?
> > 
> > 
> > Because I can't find a way to specify the time
> 
> ```
>  const errorMessage = {
>     type: 'concurrent',
>     violation_count: 20,
>     user_id: user._id,
>     prev_count: 0,
>     duration: 1000 * 60 * 60 * 24 * 7,
>   };
> 
>   await banViolation(req, res, errorMessage);
> ```
> 
> You're specifying the duration here, I believe.

Yeah, but I'm perma banning, that 'duration' doesn't work or it is being ignored by something else.

**Review by fuegovic — APPROVED** [review]
tested, ban + duration of the ban are working

File: package-lock.json
Comment (danny-avila) [inline package-lock.json:1]: please remove changes to package-lock.json

**Review by danny-avila — CHANGES_REQUESTED** [review]
please remove changes to package-lock.json

Bumping dependencies should be in a standalone PR

**Comment by Berry-13** [comment]
> please remove changes to package-lock.json
> 
> Bumping dependencies should be in a standalone PR

Sorry, I had added everything and hadn't noticed.

**Review by danny-avila — APPROVED** [review]

**Pull request merged**